farmdata2_modules/fd2_tabs/resources/ExportButtonComponent.js

```diff
@@ -0,0 +1,86 @@
+let ExportButtonComponent = {
+    template: `<div>
+                <button data-cy="download-btn" class="btn center-block btn-primary">
+                    <a style= "text-decoration: none; color: white;"
+                    :download="'tableDownload.csv'" :href="makeFile()">Download Tables</a>
+                </button>
+            </div>`,
+    props:{
+        headers: {
+            type: Array, 
+            required: true,
+        },
+        rows: {
+            type: Array,
+            required: true,
+        },
+        visibleColumns: {
+            type: Array,
+            default: null,
+        },
+        extraData: {
+            type: Array,
+            default: [],
+        },
+    },
+    methods:{
+        makeFile: function(){
+            let downloadRows = []
+            let visiableCols = this.isVisible
+            for (i = 0 ; i < this.rows.length ; i ++){
+                downloadRows[i] = ''
+                for (j = 0; j < this.rows[i].data.length ; j++){
+                    if (this.visibleColumns[j]){
+                        let column = '"' +this.rows[i].data[j] + '"'
+                        downloadRows[i] = downloadRows[i] + column + ','
+                    }
+                }
+
+            }
+            tableString = ''
+            for (k = 0 ; k < this.headers.length ; k ++){
+                if (visiableCols[k]){
+                    tableString = tableString + this.headers[k] + ','
+                }
+            }
+
+            downloadData = []
+            for(i=0; i<this.extraData.length; i++){
+                downloadData[i] = '"' + this.extraData[i].label + '","' + this.extraData[i].data + '"'
+            }
+            
+            tableString =
+                tableString +
+                '\n' +
+                downloadRows.join('\n') + 
+                '\n\n' +
+                downloadData.join('\n');
+            
+            data = new Blob([tableString], {
+                type: 'text/csv',
+            })
+            return window.URL.createObjectURL(data);
+        },
+    },
+    computed:{
+        isVisible() {
+            let updatedVis = []
+            if (this.visibleColumns == null) {
+                for (i = 0; i < this.headers.length; i++) {
+                    updatedVis.push(true);
```

Comment:
It seems like here if there are 3 headers then automatically the first 3 data columns will be displayed instead of the 3 that correspond to the locations of the headers?  For example if headers has length 3, then updatedVis will be [true, true, true].  If the downloadRows has 5 columns then just the first 3 will be displayed.  

But maybe I'm missing something.  Perhaps the downloadRows are also already pruned in the Prop?  If that is the case, then maybe this part isn't necessary at all?